### PR TITLE
docs: refine development document

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
Signed-off-by: xiang <xiang13225080@163.com>

same with https://github.com/chaos-mesh/chaos-mesh/pull/890

### What problem does this PR solve?

1. Update the development document, only some minor update
2. Add a new document about doing chaos in `chaos-daemon`

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
@fewdan @dcalvin PTAL, copy from https://github.com/chaos-mesh/chaos-mesh/pull/890

File: docs/development_guides/setup_env.md [hunk 1]
```diff
@@ -34,14 +34,21 @@ Make sure you have the above prerequisites met. Now follow the steps below to pr
 
 3. Make sure [Docker](https://docs.docker.com/install/) is installed and running on your local machine.
 
+And you need to run docker registry, more detail refer to [Docker Regitry](https://docs.docker.com/registry/). Then set environment variable `DOCKER_REGISTRY` with the registry address:
```
Comment (dcalvin) [inline docs/development_guides/setup_env.md:37]: Is this an extra step? please be clear - don't just add instructions randomly following existing steps.

File: docs/development_guides/setup_env.md [hunk 2]
```diff
@@ -64,7 +71,7 @@ With the toolchain ready, you still need a local Kubernetes cluster as the deplo
 hack/kind-cluster-build.sh
 ```
 
-The above script will create a Kubernetes cluster by kind. When you don't need this cluster, you can run the following command to delete it: 
+The above script will create a Kubernetes cluster by kind. When you don't need this cluster, you can run the following command to delete it:
```
Comment (dcalvin) [inline docs/development_guides/setup_env.md:74]:
```suggestion
The above script creates a Kubernetes cluster via Kind. When you are done with it, run the following command to delete it:
```

Comment (WangXiangUSTC) [reply docs/development_guides/setup_env.md]: done

File: docs/development_guides/setup_env.md (hunk 1 above)
Comment (WangXiangUSTC) [reply docs/development_guides/setup_env.md]: I add a new step for it

File: docs/development_guides/dev_hello_world.md [hunk 3]
```diff
@@ -177,53 +178,94 @@ dashboard:
...
+    kubectl apply -f manifests/
+    ```
+
+    You can see CRD `helloworldchaos` is created:
+
+    ```log
+    customresourcedefinition.apiextensions.k8s.io/helloworldchaos.chaos-mesh.org created
+    ```
+
+    And then you can get it as follows:
```
Comment (dcalvin) [inline docs/development_guides/dev_hello_world.md:199]:
```suggestion
    Now you can get the CRD using the command below:
```

File: docs/development_guides/dev_hello_world.md [hunk 4]
```diff
@@ -177,53 +178,94 @@ dashboard:
...
+
+      ```bash
+      helm install helm/chaos-mesh --name=chaos-mesh --namespace=chaos-testing --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock
+      ```
+
+    Then you get pods from namespace `chaos-testing` to make sure chaos-mesh is installed:
+
+    ```bash
+    kubectl get pods --namespace chaos-testing -l app.kubernetes.io/instance=chaos-mesh
+    ```
```
Comment (dcalvin) [inline docs/development_guides/dev_hello_world.md:223]: Is this part of installing chaos mesh?

File: docs/development_guides/dev_hello_world.md (hunk 3 above)
Comment (WangXiangUSTC) [reply docs/development_guides/dev_hello_world.md]: done

File: docs/development_guides/dev_hello_world.md (hunk 4 above)
Comment (WangXiangUSTC) [reply docs/development_guides/dev_hello_world.md:223]: yes, just make sure it installs success

File: docs/development_guides/dev_hello_world.md [hunk 5]
```diff
@@ -177,53 +178,94 @@ dashboard:
...
+      helm install chaos-mesh helm/chaos-mesh --namespace=chaos-testing --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock
+      ```
+
+    - For helm 2.X
+
+      ```bash
+      helm install helm/chaos-mesh --name=chaos-mesh --namespace=chaos-testing --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock
+      ```
+
+    Then you get pods from namespace `chaos-testing` to make sure chaos-mesh is installed:
```
Comment (dcalvin) [inline docs/development_guides/dev_hello_world.md:219]:
```suggestion
    To verify your installation, get pods from the `chaos-testing` namespace:
```

File: docs/development_guides/dev_hello_world.md [hunk 6]
```diff
@@ -177,53 +178,94 @@ dashboard:
...
+
+    ```bash
+    kubectl get pods --namespace chaos-testing -l app.kubernetes.io/instance=chaos-mesh
+    ```
+
+    > **Note:**
+    >
+    > The arguments `--set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock` is used to to support network chaos on kind.
```
Comment (dcalvin) [inline docs/development_guides/dev_hello_world.md:227]:
```suggestion
    > Arguments `--set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock` are used to to support network chaos on kind.
```

File: docs/development_guides/dev_hello_world.md [hunk 7]
```diff
@@ -177,53 +178,94 @@ dashboard:
...
+    2020-09-07T09:21:29.301Z        INFO    controllers.HelloWorldChaos     Hello World!    {"reconciler": "helloworldchaos"}
+    2020-09-07T09:21:29.308Z        DEBUG   controller-runtime.controller   Successfully Reconciled {"controller": "helloworldchaos", "request": "chaos-testing/hello-world"}
+    ```
 
-   > **Note:**
-   >
-   > `{pod-post-fix}` is a random string generated by Kubernetes, you can check it by executing `kubectl get po -n chaos-testing`.
+    > **Note:**
+    >
+    > `{pod-post-fix}` is a random string generated by Kubernetes, you can check it by executing `kubectl get pod -n chaos-testing`.
```
Comment (dcalvin) [inline docs/development_guides/dev_hello_world.md:267]:
```suggestion
    > `{pod-post-fix}` is a random string generated by Kubernetes. You can check it by executing `kubectl get pod -n chaos-testing`.
```

File: docs/development_guides/setup_env.md [hunk 8]
```diff
@@ -34,14 +34,21 @@ Make sure you have the above prerequisites met. Now follow the steps below to pr
 
 3. Make sure [Docker](https://docs.docker.com/install/) is installed and running on your local machine.
 
-4. Make sure `${GOPATH}/bin` is in your `PATH`.
+4. Make sure Docker Registry is running, more detail refer to [Docker Regitry](https://docs.docker.com/registry/). Then set environment variable `DOCKER_REGISTRY` with the registry address:
```
Comment (dcalvin) [inline docs/development_guides/setup_env.md:37]:
```suggestion
4. Make sure [Docker Regitry](https://docs.docker.com/registry/) is running. Set the environment variable `DOCKER_REGISTRY` with the registry address:
```

File: docs/development_guides/dev_hello_world.md (hunk 5 above)
Comment (WangXiangUSTC) [reply docs/development_guides/dev_hello_world.md]: done

File: docs/development_guides/dev_hello_world.md (hunk 6 above)
Comment (WangXiangUSTC) [reply docs/development_guides/dev_hello_world.md]: done

File: docs/development_guides/dev_hello_world.md (hunk 7 above)
Comment (WangXiangUSTC) [reply docs/development_guides/dev_hello_world.md]: done

File: docs/development_guides/setup_env.md (hunk 8 above)
Comment (WangXiangUSTC) [reply docs/development_guides/setup_env.md]: done

File: docs/development_guides/add_chaos_daemon.md [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+---
+
+In [Develop a new chaos](dev_hello_world.md), we have added a new chaos type named `HelloWorldChaos`, which will print `hello world` in `chaos-controller-manager`. To actually run the chaos, we need to configure some facilities for Chaos Daemon - so that `controller-manager` can select the specified Pods according to the chaos configuration and sends the chaos request to the `chaos-daemon` corresponding to these Pods. Once these are done, the `chaos-daemon` could run the chaos at last.
+
+This guide covers the following steps:
+
+- [Add selector for HelloWorldChaos](#add-selector-for-helloworldChaos)
+- [Implement the gRPC interface](#implement-grpc-interface-for-chaos-request)
+
+And then we can [verify your chaos](#verify-your-chaos) to verify whether this chaos works well.
```
Comment (dcalvin) [inline docs/development_guides/add_chaos_daemon.md:14]:
```suggestion
-  [Verify your chaos](#verify-your-chaos)
```

**Review by dcalvin — APPROVED** [review]
LGTM. Let's merge it for now.

**Review by fewdan — APPROVED** [review]
LGTM

**Comment by WangXiangUSTC** [comment]
@dcalvin @fewdan thanks for the review

**Pull request merged**